ui/page/wallets/restore_page.go

```diff
@@ -62,6 +62,7 @@ type Restore struct {
 
 	seedEditors       seedEditors
 	keyEvent          chan *key.Event
+	onChangeCaretPos  int // caret position when seed editor is switched
 	caretPosition     int // current caret position
```

Comment:
I believe these 2 have same function from the use case also, you have a method `caretPositionChanged()`. Why not leverage in this to carry out the change state. 

It leaves too many seemingly redundant types.